### PR TITLE
refactor: consolidate platform ts

### PR DESCRIPTION
**Pull request by rarkins**

### CODE REVIEW

File: lib/platform/gitlab/index.ts [hunk 1]
```diff
@@ -38,18 +38,16 @@ export async function initPlatform({
...
-    api.setBaseUrl(res.endpoint);
-    defaults.endpoint = res.endpoint;
+    defaults.endpoint = endpoint.replace(/\/?$/, '/'); // always add a trailing slash
+    api.setBaseUrl(defaults.endpoint);
   } else {
-    res.endpoint = defaults.endpoint;
-    logger.info('Using default GitLab endpoint: ' + res.endpoint);
+    logger.info('Using default GitLab endpoint: ' + defaults.endpoint);
   }
+  let gitAuthor;
```
Comment (ViceIce) [inline lib/platform/gitlab/index.ts]: `let gitAuthor: string;`

File: lib/platform/github/index.ts [hunk 2]
```diff
@@ -82,42 +82,47 @@ export async function initPlatform({
...
 
-  const res: PlatformConfig = {} as any;
   if (endpoint) {
     defaults.endpoint = endpoint.replace(/\/?$/, '/'); // always add a trailing slash
     api.setBaseUrl(defaults.endpoint);
   } else {
     logger.info('Using default github endpoint: ' + defaults.endpoint);
   }
-  res.endpoint = defaults.endpoint;
+  let gitAuthor;
```
Comment (ViceIce) [inline lib/platform/github/index.ts]: add type

File: test/platform/azure/index.spec.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import is from '@sindresorhus/is';
 import * as _hostRules from '../../../lib/util/host-rules';
-import { InitRepoConfig } from '../../../lib/platform/common';
+import { RepoParams } from '../../../lib/platform/common';
```
Comment (ViceIce) [inline test/platform/azure/index.spec.ts]: Maybe we should add `export * from './common'` to `platform/index.ts`, then we can reduce the nested import.

`import { RepoParams } from '../../../lib/platform';`

**Review by viceice — CHANGES_REQUESTED** [review]
Some small changes.

File: test/platform/azure/index.spec.ts (hunk 3 above)
Comment (rarkins) [reply test/platform/azure/index.spec.ts]: I don't think we import from platform/index though?

File: lib/platform/github/index.ts [hunk 4]
```diff
@@ -82,42 +82,47 @@ export async function initPlatform({
...
-  const res: PlatformConfig = {} as any;
   if (endpoint) {
     defaults.endpoint = endpoint.replace(/\/?$/, '/'); // always add a trailing slash
     api.setBaseUrl(defaults.endpoint);
   } else {
     logger.info('Using default github endpoint: ' + defaults.endpoint);
   }
-  res.endpoint = defaults.endpoint;
+  let gitAuthor: string;
+  let renovateUsername;
```
Comment (ViceIce) [inline lib/platform/github/index.ts]: Sorry, missed that. Here the type is miisng too.

File: test/platform/azure/index.spec.ts (hunk 3 above)
Comment (ViceIce) [reply test/platform/azure/index.spec.ts]: The `/index` is optional 😄

**Pull request merged**